### Задача 1

*Из колоды в 52 карты извлекаются случайным образом 4 карты.*
##### <u> a) Найти вероятность того, что все карты – крести.</u>

P = 4 любых крести из всех крести/ 4 любых карты из всей колоды.  
Всего крестей $\frac{52}{4} = 13$

Осталось записать сочетания, 
$$
P=\frac{C^{4}_{13}}{C^{4}_{52}} 
$$
и дать питону посчитать факториалы.


In [53]:
from math import factorial as f

C4_13= f(13)/(f(4)*f(13-4))
print(int(C4_13))

C4_52 = f(52)/(f(4)*f(52-4))
print(int(C4_52))

P1a = C4_13/C4_52
print(f'{P1a:%}')


715
270725
0.264106%


Получаем ответ в **0.264106% (сотые доли процента)**, довольно редкое сочетание

##### <u>б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.</u>

Можно высчитать все вероятности, когда у нас выпадет один, два, три или четыре туза и сложить вместе, или пойти от обратного и посчитать вероятность, что тузов вообще не будет, и вычесть её из $1$ (туз либо есть, либо нет, в сумме  =  100%).   
Для этого из колоды вынем тузы и посчитаем количество сочетаний $С^{4}_{48}$ , это будет благоприятными исходами для нашей нетузовой вероятности. Ну и возьмем посчитанные ранее все сочетания исходов 4 карт из колоды.

In [54]:
C4_48 = f(48)/(f(4)*f(48-4))
print(int(C4_48))
print(int(C4_52))
P1c_not = C4_48/C4_52
print(f'{(1-P1c_not):}')


194580
270725
0.2812632745405854


**28.126327%** шанс получить в руки туза, вытягивая 4 карты.


---

Здесь, не буду скрывать, сначала посчитал неправильно и только сверяясь с ребятами понял ошибку. И если,используя идею пойти от обратного события,  прийти к схожему ответу получилось, то как правильно через сочетания записать именно положительную вероятность туза я не сразу догадался.Ниже моя первая попытка решения и её доработка напильником :)  

---

Задача похожая, но нужно взять сочетание 1 любого туза из 4-х тузов и одновременно 3 любых карты из оставшейся колоды .
$$
P = \frac{C^{1}_{4} * C^{3}_{48}}{C^{4}_{52}}
$$

In [55]:
C1_4 = f(4)/(f(1)*f(4-1))
print(int(C1_4))

C3_48 = f(48)/(f(3)*f(48-3))
print(int(C3_48))
P1t = C1_4*C3_48/C4_52
print(f'{P1t:%}')

4
17296
25.555084%


замечаем, что куда-то делись 3% . Это неучтенная вероятность получить 2, 3 или 4 туза одновременно. Нужно их добавить, пока у нас только вероятность где в руке окажется ТОЛЬКО 1 туз.

Тогда придется расписывать отдельно сочетания по каждому случаю и их добавить к ответу выше:
1. $$P_{\text{2 туза}} = \frac{C^{2}_{4} * C^{2}_{48}}{C^{4}_{52}}$$
2. $$P_{\text{3 туза}} = \frac{C^{3}_{4} * C^{1}_{48}}{C^{4}_{52}}$$
3. $$P_{\text{4 туза}} = \frac{1*1}{C^{4}_{52}}$$

Пропишу по быстрому свою функцию в питоне, чтобы формулу не переписывать каждый раз.

In [56]:
def Combination(a:int,b:int) -> float:
    from math import factorial as f
    try:
        answer = f(b)/(f(a)*f(b-a))
        return answer
    except:
        print("не могу взять факториал отрицательного числа")

C = Combination


In [57]:
P2t = C(2,4)*C(2,48)/C4_52
print(f'2 туза: {P2t:%}')
P3t = C(3,4)*C(1,48)/C4_52
print(f'3 туза: {P3t:%}')
P4t = 1/C4_52
print(f'4 туза: {P4t:%}')
print(f'хотя бы один туз: {P1t+P2t +P3t+ P4t:%}')


2 туза: 2.499954%
3 туза: 0.070921%
4 туза: 0.000369%
хотя бы один туз: 28.126327%


вот теперь ответы совпали, ура! Но без помощи питона это было бы очень долгое и муторное вычисление на бумаге.  

<u>Вопрос - можно ли это было сделать как-то по другому(кроме относительной вероятности, там тоже долго),например, через размещения с повторениями? </u>

### Задача 2
*На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?*

$$
P = \frac{\text{один правильный код}} {\text{число всех комбинаций по 3 цифры на кодовом замке}} 
$$
Важный момент, что цифры прожимаются одновременно, значит порядок не важен, важно только их сочетание. Что и наталкивает на мысль применить формулу сочетаний.
Таким образом, формула вероятности будет выглядеть так:

$$
P = \frac{1}{C^{3}_{10}}
$$


In [58]:
print(int(C(3,10)))
P2 = 1/C(3,10)
print(f'{P2:%}')

120
0.833333%


**Ответ** : вероятность $\frac{1}{120}$ или шанс с первой попытки угадать **0.83%** (чуть меньше 1 процента)

### Задача 3
*В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?*

Задача похожа на семинарскую с хорошими и бракованными изделиями.  
$$
P = \frac{\text{все сочетания 3х деталей из множества крашеных}}{\text{число сочетаний 3х любых деталей из всего множества 15}}
$$
$$
P = \frac{C^{3}_{9}}{C^{3}_{15}} 
$$ 

In [59]:
#уже использую свою функцию для расчетов
print(int(C(3,9)))
print(int(C(3,15)))
P3 = C(3,9)/C(3,15)
print(f'{P3:}')

84
455
0.18461538461538463


Ответ: шанс получить 3 крашеных детали равен **18.46%**

### Задача 4
*В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?*

Здесь можно просто пойти через относительную вероятность, так как всего 2 события:
* (1й) выпадает выигрышный билет
* (2й), выпадает второй выигрышный билет из оставшихся.  
События получаются зависимыми и совместными:
$$
P(A*B)=P(A)*P(B|A)=P(1й)*P(2й|1й) = \frac{2}{100}*\frac{1}{99} = 0.0002
$$

In [60]:
P4 = (2/100)*(1/99)
print(f'{P4:}')

0.00020202020202020205


Шанс , что оба билета будут выигрышными **0,02%** (две тысячные доли процента)